### PR TITLE
Add Textual Inversion fine-tuning script

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

As per title.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: examples/stable-diffusion/README.md [hunk 1]
```diff
@@ -84,3 +84,111 @@ python text_to_image_generation.py \
 > There are two different checkpoints for Stable Diffusion 2:
 > - use [stabilityai/stable-diffusion-2-1](https://huggingface.co/stabilityai/stable-diffusion-2-1) for generating 768x768 images
 > - use [stabilityai/stable-diffusion-2-1-base](https://huggingface.co/stabilityai/stable-diffusion-2-1-base) for generating 512x512 images
+
+
```
Comment (libinta) [inline examples/stable-diffusion/README.md:88]: should we have it's own folder such as example/textual_inversion?

File: examples/stable-diffusion/textual_inversion.py [hunk 2]
```diff
@@ -0,0 +1,894 @@
...
+        if not torch.distributed.is_initialized():
+            torch.distributed.init_process_group(backend="hccl", rank=rank, world_size=world_size)
+            logger.info("Enabled distributed run.")
+    else:
+        logger.info("Single node run.")
+
+    gaudi_config = GaudiConfig.from_pretrained(args.gaudi_config_name)
+    logger.info(f"Gaudi configuration: {gaudi_config}")
+
+    if gaudi_config.use_habana_mixed_precision:
```
Comment (libinta) [inline examples/stable-diffusion/textual_inversion.py:542]: change to autocast related.

File: examples/stable-diffusion/textual_inversion.py [hunk 3]
```diff
@@ -0,0 +1,894 @@
...
+    else:
+        transformers.utils.logging.set_verbosity_error()
+        diffusers.utils.logging.set_verbosity_error()
+
+    # Log on each process the small summary:
+    logger.warning(
+        f"Process rank: {args.local_rank}, device: {device}, "
+        + f"distributed training: {bool(args.local_rank != -1)}, "
+        + f"mixed-precision training: {gaudi_config.use_habana_mixed_precision}"
+    )
```
Comment (libinta) [inline examples/stable-diffusion/textual_inversion.py:571]: change to gaudi_config.use_habana_mixed_precision

File: examples/stable-diffusion/textual_inversion.py [hunk 4]
```diff
@@ -0,0 +1,894 @@
...
+        args.lr_scheduler,
+        optimizer=optimizer,
+        num_warmup_steps=args.lr_warmup_steps * args.gradient_accumulation_steps,
+        num_training_steps=args.max_train_steps * args.gradient_accumulation_steps,
+    )
+
+    # For mixed precision training we cast the text_encoder and vae weights to half-precision
+    # as these models are only used for inference, keeping weights in full precision is not required.
+    weight_dtype = torch.float32
+    if gaudi_config.use_habana_mixed_precision:
```
Comment (libinta) [inline examples/stable-diffusion/textual_inversion.py:724]: if gaudi_config.use_torch_autocast:

File: examples/stable-diffusion/textual_inversion.py [hunk 5]
```diff
@@ -0,0 +1,894 @@
...
+
+    for epoch in range(first_epoch, args.num_train_epochs):
+        text_encoder_wrapped.train()
+        for step, batch in enumerate(train_dataloader):
+            if t0 is None and global_step == args.throughput_warmup_steps:
+                t0 = time.perf_counter()
+            batch = to_device_dtype(batch, target_device=device)
+
+            # Convert images to latent space
+            latents = vae.encode(batch["pixel_values"]).latent_dist.sample().detach()
```
Comment (libinta) [inline examples/stable-diffusion/textual_inversion.py:775]: latents = vae.encode(batch["pixel_values"].to(dtype=weight_dtype)).latent_dist.sample().detach()

File: examples/stable-diffusion/textual_inversion.py [hunk 6]
```diff
@@ -0,0 +1,1020 @@
...
+    args.num_train_epochs = math.ceil(args.max_train_steps / num_update_steps_per_epoch)
+
+    # We need to initialize the trackers we use, and also store our configuration.
+    # The trackers initializes automatically on the main process.
+    if accelerator.is_main_process:
+        accelerator.init_trackers("textual_inversion", config=vars(args))
+
+    # Train!
+    total_batch_size = args.train_batch_size * accelerator.num_processes * args.gradient_accumulation_steps
+
```
[the remark this answers is not in the document]
Comment (regisss) [reply examples/stable-diffusion/textual_inversion.py:789]: That's already the case as logging levels are different for the main process: https://github.com/huggingface/optimum-habana/blob/4dac4c42b78fdb1ad7553062202b6823c6ef6519/examples/stable-diffusion/textual_inversion.py#L607-L612
